packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -445,10 +446,65 @@ class _BoxDecorationPainter extends BoxPainter {
 
   void _paintBackgroundColor(Canvas canvas, Rect rect, TextDirection? textDirection) {
     if (_decoration.color != null || _decoration.gradient != null) {
-      _paintBox(canvas, rect, _getBackgroundPaint(rect, textDirection), textDirection);
+      // When border is filled, the rect is reduced to avoid anti-aliasing
+      // rounding error leaking the background color around the clipped
+      // shape.
+      // https://github.com/flutter/flutter/issues/13675
```

Comment:
nit: no need to link to the issue, just put whatever is relevant in the comment. (The code will probably outlive the bug database.)